From 3ccaa2a80feabbf973f2dd6d6345b50858ccf576 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Sat, 21 Dec 2024 19:24:39 -0300
Subject: [PATCH 9/9] dumper: Do not use Bash script with MSVC

Visual Studio install paths have spaces, so they should be escaped.

But in any case, it's safer to just skip the Bash wrapper.
---
 giscanner/dumper.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/giscanner/dumper.py b/giscanner/dumper.py
index cdb5400d..44d53d89 100644
--- a/giscanner/dumper.py
+++ b/giscanner/dumper.py
@@ -271,7 +271,7 @@ class DumpCompiler(object):
             sys.stdout.flush()
 
         msys = os.environ.get('MSYSTEM', None)
-        if msys:
+        if msys and not self._compiler.check_is_msvc():
             shell = os.environ.get('SHELL', 'sh.exe')
             # Create a temporary script file that
             # runs the command we want
@@ -282,6 +282,7 @@ class DumpCompiler(object):
                 if not isinstance(fcontents, bytes):
                     fcontents = fcontents.encode('utf-8')
                 f.write(fcontents)
+                print(fcontents)
             shell = utils.which(shell)
             args = [shell, tf_name.replace('\\', '/')]
         try:
@@ -289,7 +290,7 @@ class DumpCompiler(object):
         except subprocess.CalledProcessError as e:
             raise LinkerError(e)
         finally:
-            if msys:
+            if msys and not self._compiler.check_is_msvc():
                 os.remove(tf_name)
             dll_dirs.cleanup_dll_dirs()
 
-- 
2.47.0.windows.2

